src/System.Private.CoreLib/src/System/Environment.CoreRT.Unix.cs

```diff
@@ -13,23 +14,70 @@ namespace System
     public static partial class Environment
     {
         internal static int CurrentNativeThreadId => ManagedThreadId.Current;
+
+        private static object _environmentLock = new object();
```

Comment:
Nit: The separate lock object is not necessary here. You can lock on `_environment`